packages/calypso-e2e/src/lib/pages/cart-checkout-page.ts

```diff
@@ -42,4 +73,105 @@ export class CartCheckoutPage {
 		await this.page.click( selectors.removeCartItemButton( cartItemName ) );
 		await this.page.click( selectors.modalContinueButton );
 	}
+
+	/**
+	 * Obtains the text content of the payment button.
+	 *
+	 * @returns {string} Content of the payment button.
+	 */
+	async getPaymentButtonText(): Promise< string > {
+		const elementHandle = await this.page.waitForSelector( selectors.paymentButton );
+		return await elementHandle.innerText();
+	}
+
+	/**
+	 * Apply a coupon.
+	 *
+	 * If a coupon code is supplied as parameter, this method will apply the supplied code.
+	 * Otherwise, this method will apply the default coupon code found in the configuration.
+	 *
+	 * @param {string} coupon Custom coupon code.
+	 */
+	async enterCouponCode( coupon?: string ): Promise< void > {
+		await this.page.click( selectors.couponCodeInputButton );
+
+		const couponCode =
+			coupon !== undefined ? coupon : ( DataHelper.config.get( 'testCouponCode' ) as string );
```

Comment:
My thought here against what I've done, is that this method should not be responsible for looking up a coupon code if not passed in.

Reply:
Alternatively, what if we expect the test to feed in the coupon code?